File: python/7 kyu/424.py
# https://www.codewars.com/kata/5a433c7a8f27f23bb00000dc/train/python

# Passed

def split_by_value(k, elements):
    left_arr, mid_arr, right_arr = [], [], []
    for elem in elements:
        if elem == k and len(mid_arr) != 0:
            mid_arr.append(elem)
        elif elem < k:
            left_arr.append(elem)
        else:
            right_arr.append(elem)
            
    result = left_arr + mid_arr + right_arr
    return result

output = split_by_value(6, [6, 4, 10, 10, 6])
print(output)

File: nodejs/7 kyu/424.js
// https://www.codewars.com/kata/54fb963d3fe32351f2000102/train/javascript

// Passed

function collatz(n) {
  let result = 1;
  while (n !== 1) {
    if (n % 2 === 0) {
      n = n / 2;
    } else {
      n = n * 3 + 1;
    }
    result++;
  }

  return result;
}

const output = collatz(15);
console.log(output);
